sdk/cosmosdb/cosmos/src/queryExecutionContext/QueryRangeMapping.ts

```diff
@@ -0,0 +1,233 @@
+  epkMin?: string,
+  epkMax?: string,
+): ExtendedPartitionKeyRange {
+  return {
+    ...partitionKeyRange,
+    epkMin: epkMin || partitionKeyRange.minInclusive,
+    epkMax: epkMax || partitionKeyRange.maxExclusive,
+  };
+}
+
+/**
+ * @hidden
+ * Checks if a partition key range has EPK boundaries defined
+ * @param partitionKeyRange - The partition key range to check
+ * @returns True if EPK boundaries are defined
+ */
+export function hasEpkBoundaries(partitionKeyRange: ExtendedPartitionKeyRange): boolean {
+  return !!(partitionKeyRange.epkMin && partitionKeyRange.epkMax);
+}
+
+/**
+ * @hidden
+ * Gets the effective minimum boundary for a partition key range
+ * Falls back to minInclusive if epkMin is not defined
+ * @param partitionKeyRange - The partition key range
+ * @returns The effective minimum boundary
+ */
+export function getEffectiveMin(partitionKeyRange: ExtendedPartitionKeyRange): string {
+  return partitionKeyRange.epkMin || partitionKeyRange.minInclusive;
+}
+
+/**
+ * @hidden
+ * Gets the effective maximum boundary for a partition key range
+ * Falls back to maxExclusive if epkMax is not defined
+ * @param partitionKeyRange - The partition key range
+ * @returns The effective maximum boundary
+ */
+export function getEffectiveMax(partitionKeyRange: ExtendedPartitionKeyRange): string {
+  return partitionKeyRange.epkMax || partitionKeyRange.maxExclusive;
+}
+
+/**
+ * @hidden
+ * Checks if two partition key ranges overlap based on their effective boundaries
+ * @param range1 - First partition key range
+ * @param range2 - Second partition key range
+ * @returns True if the ranges overlap
+ */
+export function partitionRangesOverlap(
+  range1: ExtendedPartitionKeyRange,
+  range2: ExtendedPartitionKeyRange,
+): boolean {
+  const range1Min = getEffectiveMin(range1);
+  const range1Max = getEffectiveMax(range1);
+  const range2Min = getEffectiveMin(range2);
+  const range2Max = getEffectiveMax(range2);
+
+  return range1Min < range2Max && range2Min < range1Max;
+}
+
+/**
+ * @hidden
+ * Creates a partition key range for split scenarios
+ * @param originalRange - The original partition that is being split
+ * @param newId - ID for the new partition
+ * @param newMinInclusive - New logical minimum boundary
+ * @param newMaxExclusive - New logical maximum boundary
+ * @param epkMin - Effective partition key minimum
+ * @param epkMax - Effective partition key maximum
+ * @returns New extended partition key range for split scenario
+ */
+export function createPartitionKeyRangeForSplit(
+  originalRange: ExtendedPartitionKeyRange,
+  newId: string,
+  newMinInclusive: string,
+  newMaxExclusive: string,
+  epkMin?: string,
+  epkMax?: string,
+): ExtendedPartitionKeyRange {
+  return {
+    id: newId,
+    minInclusive: newMinInclusive,
+    maxExclusive: newMaxExclusive,
+    ridPrefix: originalRange.ridPrefix, // Inherit from parent
+    throughputFraction: originalRange.throughputFraction / 2, // Split throughput
```

Comment:
This is incorrect. It could split into more than 2 ranges.

Reply:
addressed